web/templates/layout/app.html.eex

```diff
@@ -31,6 +31,9 @@
     <meta http-equiv="X-UA-Compatible" content="IE=edge">
 
     <link rel="stylesheet" href="<%= static_path(@conn, "/css/app.css") %>">
+    <%= if assigns[:canonical_url] do %>
+      <link rel="canonical" href="<%= @canonical_url %>">
+    <% end %>
```

Comment:
I believe we should be able to use `post_url(@conn, :show, @post)` here for the full URL with slugs.